### PR TITLE
Add README.md with basic documentation for qtile.org

### PR DESCRIPTION
**Pull request by unsungNovelty**
@tych0 - I have added a README.md with basic documentation to get the site up and running. Let me know if any changes are necessary

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+- NodeJS
+- NPM
+
+## Setting up qtile.org locally
+- Install Hugo, nodejs and npm on your computer 
+- Fork and clone [qtile.org repository](https://github.com/qtile/qtile.org) locally to your computer
+- Go to `qtail` directory. `cd qtile.org/themes/qtail`
+- Run `npm install` in your terminal. This will install Tailwind CSS and it's dependencies necessary for qtile.org.
+- Go to the main website directory `/qtile.org` and run `hugo server` or `hugo server --disableFastRender`
+- Run `npm run watch-qtail` on one terminal in `themes/qtail` and `hugo server --disableFastRender` in another terminal window/tab for watching and generating file changes on the fly. This is the ideal environment for making changes.
```
Comment (tych0) [inline README.md:17]: Does this mean we have more generated code committed in the `themes/qtail` directory? Can you delete that and add build steps to the github action to regenerate it?

File: README.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+![qtile-local-env](https://github.com/unsungNovelty/qtile.org/assets/11845908/3ebda6b9-1d22-4685-9cf1-79f5f81b83ef)
```
Comment (tych0) [inline README.md:18]: Can you put this screenshot in this repo? Or alternatively just drop it, I don't know that it adds much here.

**Comment by unsungNovelty** [comment]
> Does this mean we have more generated code committed in the themes/qtail directory?

@tych0 - I didn't understand the question exactly. But there is a clear separation of code and content. `themes/qtail` has all the code logic and styling that powers the site. With contents like md files and images etc being in the parent `/qtile.org` directory. This is the documentation I was talking about explaining all this. I haven't had the time to work on it this week. But I intend to start this weekend.

> Can you delete that and add build steps to the github action to regenerate it?

Let me dig through github actions and figure out how to do that.

> Can you put this screenshot in this repo? Or alternatively just drop it, I don't know that it adds much here.

Yes. I will delete it. The screenshot was intended at new programmers who will be confused about change directory commands like `cd /qtile.org/themes/qtail`.

**Comment by tych0** [comment]
> I didn't understand the question exactly. But there is a clear separation of code and content. themes/qtail has all the code logic and styling that powers the site. 

I mean the `node_modules` directory is generated based on `package.json`, so we don't need to commit it. See 8e67ca4e590e31fda37423f5994866d3b865fb5d, but I think we can drop this as well and put a `npm install` in the makefile.

**Comment by unsungNovelty** [comment]
> See https://github.com/qtile/qtile.org/commit/8e67ca4e590e31fda37423f5994866d3b865fb5d, but I think we can drop this as well and put a npm install in the makefile

Hmmm. I am not sure Github actions. So I might be wrong here. But why have you added `/public` and `/resources` directories to `.gitignore`? 

`/public` contains _THE website_ output which is needed to host the website. And `resources` contains the images that is needed for thumbnails of screenshots. I think there are duplicates in those thumbnails which should be removed. But not sure the entire directory should be ignored.

**Comment by tych0** [comment]
Yes, we do not want to commit the generated code to the repository, since it will just add bloat years down the line when we have generated lots of different versions of it.

Instead, I set it up to push the output of the website (i.e. exactly these public and resource directories) to a separate qtile.github.io, which we can delete/refresh at will when it gets too big (if anyone bothers to clone it at all).

I didn't realize at the time that there was a node_modules directory that was also autogenerated. But I don't think we should do that either.

I think a good patch would be: delete the node_modules directory, and add whatever the necessary npm install actions are to re-create it during the github actions run.

**Comment by unsungNovelty** [comment]
@tych0 - I misunderstood your question earlier. The only generated file inside `/themes/qtail` directory is the `style.css`. I have added the file to `.gitignore` and have tried to make the changes necessary in the makefile. I am new to makefile, so can you please have a look at it?

File: Makefile [hunk 3]
```diff
@@ -1,5 +1,8 @@
+all:
+	cd themes/qtail/; 
+	npm run build-qtail;
 public:
-	hugo --gc --minify
+	hugo --gc --minify;
```
Comment (tych0) [inline Makefile:5]: no need for semicolons here

File: Makefile [hunk 4]
```diff
@@ -1,5 +1,8 @@
...
 
 clean:
-	rm -rf public resources
+	rm -rf public resources themes/qtail/assets/style.css
```
Comment (tych0) [inline Makefile:8]: Can you also delete the checked-in copy in this commit?

And it looks like we'll probably need to change the github actions code to just do a `make`, so we don't have to encode this twice. That's in .github/workflows/... can you make that change in this commit as well?

**Comment by unsungNovelty** [comment]
@tych0 - Can you please have a look and see if this looks OK?
Excuse multiple commits. First time using github actions and yaml file. Had to rely on my spidey senses to figure it out.

**Comment by tych0** [comment]
The github ci stuff looks good, but the other comments I made still need to be fixed it looks like. I can squash the series once that's all done, or you can.

**Comment by unsungNovelty** [comment]
@tych0 - I have removed the screenshot from readme.md and have made the changes to `Makefile`. Did I miss anything?

**Comment by tych0** [comment]
Hm, I don't understand what I did exactly to close this, but now I've pushed these to master and it looks like we've had a successful deploy: https://github.com/qtile/qtile.org/actions

Thanks!

**Comment by unsungNovelty** [comment]
@tych0 - Great! Was waiting for this to be merged. I will start working on the rest of the changes soon.

**Pull request closed without merging**